source/gui/startupDialogs.py

```diff
@@ -104,7 +105,26 @@ def __init__(self, parent):
 	def onOk(self, evt):
 		layout = self.kbdNames[self.kbdList.GetSelection()]
 		config.conf["keyboard"]["keyboardLayout"] = layout
-		config.conf["keyboard"]["useCapsLockAsNVDAModifierKey"] = self.capsAsNVDAModifierCheckBox.IsChecked()
+		NVDAKeysVal = (
+			(NVDAKey.CAPS_LOCK.value if self.capsAsNVDAModifierCheckBox.IsChecked() else 0)
+			| (config.conf["keyboard"]["NVDAModifierKeys"] & NVDAKey.NUMPAD_INSERT.value)
+			| (config.conf["keyboard"]["NVDAModifierKeys"] & NVDAKey.EXTENDED_INSERT.value)
+		)
+		if NVDAKeysVal == 0:
+			log.debugWarning("No NVDA key set")
+			gui.messageBox(
+				_(
+					# Translators: The title of an error message box displayed when validating the startup dialog
+					"Despite your choice, caps lock will remain used as NVDA modifier key"
+					" since no other possible key is currently configured as NVDA modifier key."
```

Comment:
I've suggested a rephrase of this error message to be more clear what the choice was, and why it failed

```suggestion
					"No NVDA modifier key has been selected. "
					"As an NVDA modifier key must be set, caps lock will be saved as the NVDA modifier key. "
```

Reply:
I have opted for another phrasing in 61af5f252be7acb40e51cb7aef5f0992a90b3e79 since I do not think that NVDAKey configuration should be modified in this case.